### PR TITLE
Accessibility: Improve HelpModal markup

### LINKED ISSUE #79516: [accessibility issue] HelpModal keyboard shortcut tables are difficult to understand for screen reader users
**Steps to reproduce**

Open up the help modal with either `h` ([an issue filed in itself](https://github.com/grafana/grafana/issues/74841)) or using `shift + ?` to open the help modal. It shows four html tables with keyboard shortcuts categorised inside them.

**The problems**

"Global", "Dashboard", "Focused Panel", and "Time Range" are marked up as the table headers (spanning two columns), however they do not serve as the table headers of the table (instead they serve as the caption of the table).

No tables have a programmatically associated label or accessible name:

The tables in the keyboard shortcuts dialog do not have accessible names. Instead the visible labels for the tables are marked up as the table headers for each table.

Screenshot of Grafana with the Help dialog open with the VoiceOver caption box beside it. It shows that the first item in the table doesn't have a header associated with it.

UHG Audit IDs:
1.3112
1.3114

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Removes the table in the HelpModal to make it more accessible for screen readers. With the help of the `Text` component, the contrast issue for the light theme is addressed. Also made it readable for extra small devices, because why not?

Before:
Screenshot 2024-02-21 at 14 29 13
After:
image

**Why do we need this feature?**

It is very heard to read for screenreaders.
 
**Who is this feature for?**

Screen reader users

**Which issue(s) does this PR fix?**:

Fixes #79516 

**Special notes for your reviewer:**

I used a definition list and tried it with the screen reader, but I am not convinced. It did work well with the CSS grid though.

There was a discussion about separation the keys with modifiers, but I don't think we should do it, as it highlights that they should be pressed at the same time as opposed to in succession. 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tskarhed** [comment]
Should we write/design something that makes it easier to understand if you should press the keys at the same time or in order?

**Review by ckbedwell — COMMENTED** [review]
To be honest, I'm not that precious about this but I'd probably say I'd rather have a table over the Definition List. Two main reasons why:

1. Tables actually work really well with screen readers - in particular that you can easily navigate up and down columns which I think would be pretty desirable when as a user you want to see what shortcuts are available, _then_ look at what the shortcut is if you want to use it (for this reason I'd also think about changing the design and put the command first then the keys second  -- again minor and not sure if design will sign off on that change or not but all the other sites I've spot-checked do it this way.)

I should have made this clearer in the filed issue but the current implementation's main problem is the way the table header is implemented. Neither columns have their own header, just a shared one that spans two columns, so it can be disorientating for screen reader users when entering the table.

2. I think the screen reader implementation for definition lists is pretty annoying. There is clearly 8 commands which you have marked up correctly but the screen reader reads it as "16 items", as it considers each definition term and definition description as an item and doesn't think they are related 🤷 You have to go through all of the commands and keys associated before finding out what the next command is.

> There was a discussion about separation the keys with modifiers, but I don't think we should do it, as it highlights that they should be pressed at the same time as opposed to in succession.

I never noticed that they could be hit one after the other until this comment, I wasn't sure what you meant until I tested it 😅  So on this note, the problem with them listed like this is they are treated separately with VoiceOver (when I tested with NVDA they are read together, though, so it is platform specific -- might also be because of the settings I use for my screen readers).

You could go the route of rendering a grouped text version that is hidden visually (e.g. `className="sr-only"`) and putting `aria-hidden` on the visible keys so they aren't read out twice.

Lastly -- for screen reader users `Esc` is read out as `e - s - c` rather than escape, (same with `c - m - d`). The few keys that are capitals also aren't clear to screen reader users. It might be worth specifically mentioning it is `d shift + c` as an example.

**Comment by tskarhed** [comment]
I added specific characters for command and shift. I am doing sr-only for ctrl and esc. Writing the whole word seems a bit excessive, especially since there is a convention around it.

File: public/app/core/components/help/HelpModal.tsx [hunk 1]
```diff
@@ -140,86 +143,74 @@ export interface HelpModalProps {
...
-          </div>
+      <Grid columns={{ xs: 1, sm: 2 }} gap={3}>
+        {Object.values(shortcuts).map(({ category, shortcuts }) => (
+          <section key={category}>
+            <div className={styles.categoryHeader}>
+              <Text element="h3" variant="h5">
+                {category}
+              </Text>
+            </div>
+            <dl className={styles.keysAndDescriptions}>
```
[the remark this answers is not in the document]
Comment (tskarhed) [reply public/app/core/components/help/HelpModal.tsx]: Using a `caption` for the table

**Comment by amy-super** [comment]
One little trick that might help the layout look a bit better would be to reorder the sections. So `Global` on the top left, `Dashboard` on the bottom left, `Time Range` on the top right, `Focused Panel` on the bottom right. 

I'd also change them to all to sentence case (`Focused panel`, `Time range`) to be a more consistent with Grafana styling.

**Review by ckbedwell — COMMENTED** [review]
Left one minor comment about the srOnly implementation to make it meet the official standard.

I also noticed that when I am zoomed in a bit I can't scroll up and down the content using just my keyboard. @eledobleefe [looked at this problem recently](https://github.com/grafana/design-system/pull/117/files#diff-596152d30a8b7f508b5c95c7eb146e27b695e362dbb4eda91745acbf30b17ab7R50-R54) and we agreed the best solution is to get modal authors to add `tabIndex={0}` to content containers (in this case the Grid component in the HelpModal) with no focusable elements.

One last thing I want to highlight so that if this ever gets flagged in a future audit we can say we are aware of it and have chosen to ignore this best practice (or maybe add it to be thought about in the future for these kind of designs? cc @amy-super).

[So it is best practice that when using a table, that the table headers are visible](https://dequeuniversity.com/rules/axe/4.1/empty-table-header). (Quick side node --  the way our `.sr-only` class is written Deque's automated dev tools think it is visible...). As previously discussed, this design could be implemented as a definition list, or just without table markup at all but we've chosen not to do that because we believe the experience is worse for users that way.

**Comment by tskarhed** [comment]
Okay, I've done this:
- Added tabIndex for scrolling with keyboard
- Changed the way ctrl and esc are read to screen readers

From my understanding, you are okay with the table headers not showing.

**Review by ckbedwell — APPROVED** [review]
LGTM! 🌟

**Pull request merged**